AirportTicketBookingSystem.Tests/DomainTests/ServicesTests/FlightServiceTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Moq;
+using FluentAssertions;
+using AutoFixture;
+using AirportTicketBookingSystem.Domain.Models;
+using AirportTicketBookingSystem.Domain.Repositories;
+using AirportTicketBookingSystem.Domain.Services;
+public class FlightServiceTests
+{
+    private readonly Mock<IFlightRepository> _mockFlightRepo;
+    private readonly FlightService _flightService;
+    private readonly Fixture _fixture;
+
+    public FlightServiceTests()
+    {
+        _mockFlightRepo = new Mock<IFlightRepository>();
+        _flightService = new FlightService(_mockFlightRepo.Object);
+        _fixture = new Fixture();
+    }
+
+    [Fact]
+    public void AddFlight_ShouldCallRepositoryAddFlight()
+    {
+        var flight = _fixture.Create<Flight>();
+
+        _flightService.AddFlight(flight);
+
+        _mockFlightRepo.Verify(repo => repo.AddFlight(flight), Times.Once);
+    }
+
+    [Fact]
+    public void AddFlight_ShouldThrowException_WhenFlightAlreadyExists()
```

Comment:
this shouldn't be part of the FlightService tests, since this validation is hidden in the repo method, you can't tell from the service that this AddFlight method is doing any validations. This should belong to the repo tests.